_posts/en/newsletters/2024-05-08-newsletter.md

```diff
@@ -0,0 +1,311 @@
+    Bitcoin Core for over a decade have had an orphan pool where they
+    temporarily store a limited number of transactions that were
+    received before their parents.  This compensates for the fact that,
+    on a P2P network, transactions can sometimes naturally be received
+    out of order.
+
+  - A few moments later, Alice relays the parent transaction to her
+    peer.
+
+  - Before this merged PR, the peer would notice that the parent's
+    feerate was too low and refuse to accept it; now that it had
+    evaluated the parent transaction, it would also remove
+    the child transaction from the orphan pool.  After this PR, the peer
+    notices that it has a child for the parent in its orphan pool and
+    evaluates aggregate feerate of both transactions together, allowing
+    them both into the mempool if that feerate is above its floor (and
+    if they're both otherwise acceptable according to the node's local
+    policy).
+
+  It's known that this mechanism can be defeated by an attacker.
+  Bitcoin Core's orphan pool is a circular buffer that can be added to
+  by all of its peers, so an attacker who want to prevent this type of
+  package relay can spam peers with many orphan transactions,
+  potentially leading to a fee-paying child transaction being evicted
+  before it's parent is received.  A [follow-up PR][bitcoin core #27742]
+  may give each peer exclusive access to part of the orphan pool to
+  eliminate this concern.  Additional improvements requiring changes to
+  the P2P protocol are described in [BIP331][].
+
+- [Bitcoin Core #28016][] begins waiting for all seed nodes to be polled
+  before polling DNS seeds.  Users can configure both seed nodes and DNS
+  seeds.  A seed node is a regular Bitcoin full node; Bitcoin Core can
+  open a TCP connection to the node, request a list of addresses for
+  potential peers, and close the connection.  A DNS seed returns IP
+  addresses for potential peers over DNS, allowing that information to
+  travel and be cached across the DNS network so the owner of the DNS
+  seed server doesn't learn the IP address of the client requesting the
+  information.
+
+  Before this PR, if a user configured polling of seed nodes and kept
+  the default configuration to also use DNS seeds, they would both be
+  contacted in parallel and whichever was faster would dominate the peer
+  addresses that the node would try.  Given the lower overhead of DNS
+  and that fact that results might already be cached by a server
+  physically near the user, DNS would usually win.  After this PR, seed
+  nodes are given preference under the belief that a user who sets a
+  non-default option would prefer the results of that option over
+  default results.
+
+- [Bitcoin Core #29623][] makes various improvements to warning users if
+  their local time seems to be more than 10 minutes out of sync with the
+  time of their connected peers.  A node with a bad clock might
```

Comment:
Think referencing this PR could be helpful for readers not already familiar with it:
https://bitcoinops.org/en/newsletters/2024/02/07/#bitcoin-core-28956